Analysis of ricochet flag (#9)
BZFlag-Dev/bzflag.org: _documentation/user/flags/ricochet.md

summary: Shots bounce off walls. Don't shoot yourself!
---

> The Ricochet (R) flag allows the player to fire shots that can bounce off of the objects they hit. The direction of the bounce is determined by the angle under which the platform is hit. An exception are objects with the attributes `shootthrough` or `passable`, where the result is the bullet passing through the object.

Reviewer: > The direction of the bounce is determined by the angle under which the platform is hit.

This is awkward. What platform and why am I hitting under it? I think you may be trying to allude to the [Law of Reflection](https://www.physicsclassroom.com/Class/refln/u13l1c.cfm) but it's not clear.

Author: @jwmelto valid point; will elaborate. Any concrete suggestions beforehand?

Reviewer: well, besides being somewhat obvious basic physics, "the angle of reflection is equal to the angle of incidence" comes to mind.

Author: I was asking for a concrete example here to avoid potential further displeasure in the wording, as we are basically thinking of the same thing just the wording was off.

Will use your suggestion.

> A lot of maps utilize the default ricochet option to already allow the effect of Ricochet without the tanks being required to hold the specific flag for it. The Ricochet flag holds meaning and effect on maps that do not utilize said option.

Reviewer: Maybe it's just an evolving pet peeve, but "a lot of maps" is irrelevant to the description of the flag.
```suggestion
The Ricochet flag has no value on maps that enable ricochet. For maps where ricochet is not enabled, the Ricochet flag offers the holder the advantage of banking shots.
```

Author: @jwmelto agreed, will change.

## Tactics

> Ricochet is potent on maps filled with objects, usually boxes, behind which a tank can hide from direct enemy fire, all while attempting to shoot them down by using the objects as proxies off of which own bullets bounce in hopes of hitting the opponent.

Reviewer: ```suggestion
Ricochet is potent on maps filled with objects behind which a tank can hide from direct enemy fire, while attempting to shoot the opponent by bouncing shots off objects.
```

Author: @jwmelto can you elaborate as to why you removed the mention of boxes? They are notable for having straight walls, allowing for commonly 'useful' ricochet, contrary to most pyramids. Note that this is the `Tactics` section.

Otherwise good change suggestion.

Reviewer: Ricochet is feasible and useful on maps with pyramids also. In fact, once can use ricochet off the angular side of a pyramid to hit a tank above you. I just didn't seen a need to limit the discussion to boxes.

Author: That is true; will change.

> Ricochet is potent on maps filled with objects, usually boxes, behind which a tank can hide from direct fire from the opponent, all while attempting to shoot them down by using the objects as proxies off of which own bullets bounce in hopes of hitting the opponent.

Reviewer 2: ```suggestion
Ricochet is potent on maps filled with objects, usually boxes, behind which a tank can hide from direct enemy fire, all while attempting to shoot them down by using the objects as proxies off of which own bullets bounce in hopes of hitting the opponent.
```

The two "from"s sound repetitive to me

> Efficiently utilizing the Ricochet flag requires a high amount of estimating the bullet's trajectory by planning the path before it is shot, usually through the use of radar. Complexity behind the estimation increases with each additional bounce required to reach the target.

Reviewer: "Efficiently" is borderline subjective. Consider a more declarative style
```suggestion
Effective use of the Ricochet flag requires accurate estimation of the path of a shot after it bounces off an obstacle. Multiple bounces compounds the difficulty. One may observe the trajectory of a shot and adjust the point of aim to increase accuracy.
```

Author: @jwmelto agreed, will change.

> Efficiently utilizing the Ricochet flag requires a high amount of estimating of the bullet's trajectory by planning the path before the shot, usually through the use of radar. Complexity behind the estimation increases with each additional bounce required to reach the target.

Reviewer 2: ```suggestion
Efficiently utilizing the Ricochet flag requires a high amount of estimating the bullet's trajectory by planning the path before it is shot, usually through the use of radar. Complexity behind the estimation increases with each additional bounce required to reach the target.
```

something sounded off. thoughts on this?
